eras/shelley-ma/impl/src/Cardano/Ledger/Mary/Value.hs

```diff
@@ -192,7 +223,7 @@ instance CC.Crypto crypto => Val (MaryValue crypto) where
               + repOverhead
           )
 
-  isAdaOnly (MaryValue _ v) = Map.null v
+  isAdaOnly (MaryValue _ (MultiAsset m)) = m == mempty
 
```

Comment:
why not
isAdaOnly (MaryValue _ (MultiAsset v)) = Map.null v

There is quite a bit over overhead involved using the EQ class and the Monoid class instances we can avoid.
This might appear else where, as well. Might want to search for this pattern.

Reply:
Sure.  I made an almost arbitrary choice to change it to this because I was thinking to use the higher level abstraction - but really just as a way to choose one. 
The performance overhead is a real reason to prefer the other one - I'm happy to change it and glad to have learnt this, thanks!